index.html

```diff
@@ -1028,6 +1061,461 @@ <h2>
+    </section>
+    <section>
+      <h3>
+        <dfn>GamepadHapticsResult</dfn> Enum
+      </h3>
+      <p>
+        An enum representing the result of a call to
+        {{GamepadHapticActuator}}.{{GamepadHapticActuator/playEffect()}} or
+        {{GamepadHapticActuator}}.{{GamepadHapticActuator/reset()}}.
+      </p>
+      <pre class="idl">
+        enum GamepadHapticsResult {
+          "complete",
+          "preempted"
+        };
+      </pre>
+      <dl data-dfn-for="GamepadHapticsResult">
+        <dt>
+          <dfn>complete</dfn>
+        </dt>
+        <dd>
+          <p>
+            The value the {{Promise}} returned by
+            {{GamepadHapticActuator}}.{{GamepadHapticActuator/playEffect()}}
+            or {{GamepadHapticActuator}}.{{GamepadHapticActuator/reset()}}
+            will resolve to when the effect completes normally.
+          </p>
+        </dd>
+        <dt>
+          <dfn>preempted</dfn>
+        </dt>
+        <dd>
+          <p>
+            The value the {{Promise}} returned by
+            {{GamepadHapticActuator}}.{{GamepadHapticActuator/playEffect()}}
+            or {{GamepadHapticActuator}}.{{GamepadHapticActuator/reset()}}
+            will resolve to when the operation does not complete because the
+            {{Document}} is `"hidden"` or another effect was started
+            on the same actuator.
+          </p>
+        </dd>
+      </dl>
+    </section>
+    <section>
+      <h3>
+        <dfn>GamepadHapticEffectType</dfn> Enum
+      </h3>
+      <p>
+        The effect type defines how the effect parameters are interpreted by
+        the actuator.
+      </p>
+      <p>
+        The [=user agent=] MAY modify the behavior of
+        {{GamepadHapticActuator}}.{{GamepadHapticActuator/playEffect}} to
+        extend support for new {{GamepadHapticEffectType}} values and
+        {{GamepadEffectParameter}} attributes. Implementations MUST NOT
+        modify the definition of a [=valid effect=] for effect types defined
+        in this specification.
+      </p>
+      <pre class="idl">
+        enum GamepadHapticEffectType {
+          "dual-rumble"
+        };
+      </pre>
+      <dl>
+        <dt>
+          <dfn data-for="GamepadHapticEffectType">"dual-rumble" effect
+          type</dfn>
+        </dt>
+        <dd>
+          <p>
+            {{GamepadHapticEffectType/"dual-rumble"}} describes a haptics
+            configuration with an eccentric rotating mass (ERM) vibration
+            motor in each handle of a standard gamepad. In this
+            configuration, either motor is capable of vibrating the whole
+            gamepad. The vibration effects created by each motor are unequal
+            so that the effects of each can be combined to create more
+            complex haptic effects.
+          </p>
+          <p>
+            A {{GamepadHapticEffectType/"dual-rumble"}} effect is a
+            fixed-length, constant-intensity vibration effect intended for an
```

Comment:
What does "fixed-length" mean here, and how does that play with "duration"?

Reply:
I haven't written this, but I think that fixed-length in this context means that, once an effect with a duration `d` has been issued, you cannot modify `d`. Your only options are to either preempt it or wait for it to finish.

Reply:
Ah! Ok, then maybe: 

```suggestion
            fixed-duration, constant-intensity vibration effect intended for an
```